### DISCUSSION TITLE
Pass new config to client (auth/subscriptions)

### DISCUSSION THREAD

**Discussion by Miloshinjo** [urql-graphql/urql#2100] [Q&A] [answered]
I am wondering if there is a way in urql to pass a new config to the client? 

I have one param that is sent to the backend via `headers`, and thus is needed by the `authExchange` and `subscriptionsExchange` in order to make authenticated requests. 

Is it somehow possible to change that variable during runtime of the app and have the client use the new one? 

p.s. Think of it as multi-tenant structure. To be more precise. `/:dynamicId/restOfTheApp` , I want client to use this :dynamic ID in headers each time that changes.

**Comment by JoviDeCroock** [comment, marked as answer]
You can make creation of your client part of the `React-lifecycle` which means that if `router.query.dynamicId` changes your `useMemo`/... would trigger a new client to be created and provided to your subtree.

**Reply by Miloshinjo** [reply]
Thanks for the quick answer ☺️. Would that wipe the entire cache on each new change? 

Another question, would it be better if I was to pass these params on each `useQuery` `useMutation` etc... thru `fetchOptions`?

**Reply by JoviDeCroock** [reply]
I mean, I _think_ it would probably be easier to make an abstraction around your hooks to manage this but I don't know the details of your particular case, so that is indeed an option.

Your cache _would_ reset if you make it part of that closure but if it's not part of that closure that reinstantiates it won't be wiped. So basically making a new `cacheExchange` by invoking the method will wipe the cache.

**Reply by Miloshinjo** [reply]
Thanks JoviDeCroock, you really helped me out, I will try both directions and see what I get :)
